integration-cli/docker_api_logs_test.go

```diff
@@ -82,3 +86,96 @@ func (s *DockerSuite) TestLogsAPIContainerNotFound(c *check.C) {
 	c.Assert(err, checker.IsNil)
 	c.Assert(resp.StatusCode, checker.Equals, http.StatusNotFound)
 }
+
+func (s *DockerSuite) TestLogsAPIUntilFutureFollow(c *check.C) {
+	testRequires(c, DaemonIsLinux)
+
+	name := "logsuntilfuturefollow"
+	dockerCmd(c, "run", "-d", "--name", name, "busybox", "/bin/sh", "-c", "while true; do echo hello; sleep 1; done")
+	waitRun(name)
+
+	until := daemonTime(c).Add(5)
+
+	type logOut struct {
+		out string
+		err error
+	}
+
+	client, err := request.NewClient()
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	cfg := types.ContainerLogsOptions{Until: until.Format(time.RFC3339Nano), ShowStdout: true, Timestamps: true}
+	reader, err := client.ContainerLogs(context.Background(), name, cfg)
+	c.Assert(err, checker.IsNil)
+
+	actualStdout := new(bytes.Buffer)
+	actualStderr := new(bytes.Buffer)
+	_, err = stdcopy.StdCopy(actualStdout, actualStderr, reader)
+	c.Assert(err, checker.IsNil)
+
+	chLog := make(chan logOut)
+	go func() {
```

Comment:
Since this is copying everything to a bytes.Buffer, I don't think there's any need for the goroutine here since it will return at EOF if a `\n` is not reached.